### ISSUE TITLE
Put a parameters criterion to deep models function

### ISSUE THREAD

**Issue by tgnassou** [scikit-adaptation/skada#161]
The function to create deep DA models should have a parameter `criterion` or `base_criterion` or `source_criterion`. For now, it is hardcoded as a cross-entropy loss but if the model has a softmax for example one should be able to put a BCELoss.

**Comment by Som-III** [comment]
Added a 'criterion' parameter with default as 'cross_entropy_loss'
